sdk/src/runtime/worker.tsx

```diff
@@ -278,6 +278,29 @@ export const defineApp = <
         }
 
         if (e instanceof Response) {
+          // If a route or server action returns a redirect (Location header),
+          // we can't rely on fetch+redirect:\"manual\" (browsers often make it
+          // an opaque redirect and hide headers). To ensure the client can
+          // always read the target, we convert the redirect into a 204 with an
+          // explicit instruction header the client can consume.
+          const locationHeader =
+            e.headers.get("Location") || e.headers.get("location");
+          if (locationHeader) {
+            try {
+              const absolute = new URL(locationHeader, request.url).toString();
+              const headers = new Headers(e.headers);
+              headers.delete("Location");
+              headers.delete("location");
+              headers.set("x-rwsdk-redirect-location", absolute);
+              headers.set("x-rwsdk-redirect-status", String(e.status || 303));
+              return new Response(null, {
+                status: 204,
+                headers,
+              });
```

Comment:
@harryhcs I think this will break redirects, e.g. from interruptors / middleware, for page loads. For that case, we still want to preserve the response that was thrown, since there's no client side fetch to worry about there.

One way to solve that:
1. Catch error responses thrown or returned when calling the action handler over here: https://github.com/redwoodjs/sdk/blob/a5b5ed20db8390c76740ee85dc3e21ed52db70bb/sdk/src/runtime/register/worker.ts#L99
2. Set the `x-rwsdk` headers at that point (using `requestInfo.response.headers`)
3. On the client side, look for these headers in fetch responses, and use them as a signal to redirect (we already do this in this PR)

We shouldn't need any code here anymore if we do things that way, I don't think.